internal/services/network/network_manager_deployment_resource.go

```diff
@@ -264,7 +269,12 @@ func (r ManagerDeploymentResource) Update() sdk.ResourceFunc {
 				return fmt.Errorf("creating %s: %+v", id, err)
 			}
 
-			if err = resourceManagerDeploymentWaitForFinished(ctx, client, id, metadata.ResourceData); err != nil {
+			deadline, ok := ctx.Deadline()
+			if !ok {
+				return fmt.Errorf("update context for %s had no deadline", *id)
```

Comment:
These would be internal errors, so for consistency want to be:

```suggestion
				return fmt.Errorf("internal-error: context had no deadline")
```